modules/persistence/src/services/metadata/ToC/index.ts

```diff
@@ -30,15 +38,16 @@ const isInsertionCandidateNode = (node: ToC, associated_products: AssociatedProd
   return !!(nodeHasNoChildren && nodeInProducts);
 };
 
-const mergeNode = (node: ToC, tocs: ToCInsertions) => {
+const mergeNode = (node: ToC, tocs: ToCInsertions, currentProject) => {
   // Options might be undefined, so safely cast to {} if nullish
   node.options = node.options ?? {};
+  const needsUrlifiedToC = currentProject !== node?.options?.project;
 
-  const project = tocs[node?.options?.project];
-  const branches = Object.keys(project);
+  const associatedProject = tocs[node?.options?.project];
+  const branches = Object.keys(associatedProject);
```

Comment:
when testing with an umbrella entry that does not have its associated metadata entry persisted yet, getting error for `VM506:1 Uncaught TypeError: Cannot convert undefined or null to object`

Reply:
Good catch here - I should just short circuit here in general and raise a log if we can't find a branches object.